experiments with grain property classes and functions

sgpearson17 opened the pull request:
I have made a first attempt at breaking out the basic grain size property functions used in soulsby (Dstar, shields_cr, shields_number). I have not had time to finish them all yet, and also still want to finish working on the unit tests (using example data from Soulsby (1997) textbook) but am uploading what I have for now so that @monica-aguil  and @npannozzo can take a look. I will come back to this in two weeks, as next week.

sgpearson17 committed:
last week's attempt to write Soulsby P function with classes for sediment properties

sgpearson17 committed:
experiments with grain property classes and functions

I have made a first attempt at breaking out the basic grain size property functions used in soulsby (Dstar, shields_cr, shields_number). I have not had time to finish them all yet, and also still want to finish working on the unit tests (using example data from Soulsby (1997) textbook) but am uploading what I have for now so that Monica and Natascia can take a look

npannozzo commented:
Thanks for this, @sgpearson17. @monica-aguil and I have been working on this with the help of Manuel and we now have an appropriate strategy on how to perform this task. We have already implemented the structure and will finish populating it with the content next week, after a short catch-up with the team to make sure that all the content necessary to support all the types of pre-processing linked to all the scenarios currently available is in there.

sgpearson17 reviewed (commented):
@monica-aguil and @npannozzo, here is the commit with my function tests: 26a8deef3b4d8a2f8bd2791c758e0e08e4c02e2d. feel free to copy into your own branches to work on

manuGil commented:
We will look at this once the data structure PR is merged #110

sgpearson17 committed:
documentation (old and benchmark runs)

existing documentation (old user manual and paper draft) plus new information about matlab benchmark runs

sgpearson17 commented:
Whoops, somehow my updated documentation ended up here. I think I didn't correctly update my dev branch. Something to check out later this week and fix.

sgpearson17 commented:
closed after talking with Manuel and Aysun re: cherrypicking doc commit out to another PR
